### DISCUSSION TITLE
CommandClient usage in application service crashes

### DISCUSSION THREAD

**Discussion by sudhamani-hcl** [edgexfoundry/community#189] [General]
Hi Team,

I am working on application service based on message bus trigger. In one of the pipeline functions, I am sending a command to device service using **CommandClient interface: IssueGetCommandByName** which results in crash.

Please find the below attached crash log and let us know your pointers.

[test-app-debug-log.txt](https://github.com/edgexfoundry/community/files/12884000/test-app-debug-log.txt)

Thanks,
Sudhamani

**Comment by cloudxxx8** [comment]
@sudhamani-hcl please provide the version and clear recreate steps

**Reply by sudhamani-hcl** [reply]
Hi @cloudxxx8 ,

We are using EdgeX v2.2.0 version.
From the southbound side, device-random is running and it sends random event to message bus.
This message bus event triggers the pipeline function in the application service. The executed pipeline function is SendTestCmd() as shown below.

Our observation is ctx.<_anyclient_> gives crash, eg: ctx.CommandClient() gives crash except ctx.LoggingClient().

Please find below the pipeline function used.


```
func SendTestCmd(ctx interfaces.AppFunctionContext, data interface{}) (bool, interface{}) {
        lc := ctx.LoggingClient()
        lc.Debug("**********START:SendCmdtotest function*************")
        var response interface{}
        //response, err := ctx.CommandClient().IssueGetCommandByName(context.Background(), "Random-Integer-Generator01", "RandomValue_Int16", "no", "yes")
        //response, err := ctx.CommandClient().IssueGetCommandByName(context.TODO(), "Random-Integer-Generator01", "RandomValue_Int16", "no", "yes")
        //response, err := ctx.CommandClient().AllDeviceCoreCommands(context.Background(), 0, -1)
        //response, err := ctx.CommandClient().AllDeviceCoreCommands(context.TODO(), 0, -1)
        //response, err := ctx.CommandClient().AllDeviceCoreCommands(context.Context(), 0, -1)
      //response, err := ctx.CommonClient().Ping(context.Background())

      //response, err := ctx.GeneralClient().FetchMetrics(context.Background())
        response, err := ctx.EventClient().AllEvents(context.Background(), 0, -1)
        if(err == nil) {
        }
      //ret := ctx.GetAllValues() // working
        //if(ret == nil) {
      //}

        lc.Debug("**********END:SendCmdtotest function*************")
        return true, response;
}

```
Thanks,
Sudhamani.

**Reply by cloudxxx8** [reply]
Did you set the Command client configuration like this:
https://github.com/edgexfoundry/app-service-configurable/blob/v2.3.1/res/sample/configuration.toml#L252-L256

**Comment by sudhamani-hcl** [comment]
Hi @cloudxxx8 ,

Thanks for the input.
After adding the core-data client in the configuration file it is not crashing. Hence closing the discussion.

Thanks,
Sudhamani
